lib/ui/painting.dart

```diff
@@ -3931,6 +3978,35 @@ class _ComposeImageFilter implements ImageFilter {
   int get hashCode => Object.hash(innerFilter, outerFilter);
 }
 
+class _FragmentShaderImageFilter implements ImageFilter {
+  _FragmentShaderImageFilter(this.shader);
+
+  final FragmentShader shader;
+
+  late final _ImageFilter nativeFilter = _ImageFilter.shader(this);
+
+  @override
+  _ImageFilter _toNativeImageFilter() => nativeFilter;
+
+  @override
+  String get _shortDescription => 'shader';
+
+  @override
+  String toString() => 'ImageFilter.shader()';
```

Comment:
Nit: Do we not disambiguate objects in any other way? Perhaps dump the hashcode as well?

Reply:
Nope, added hashcode from shader object.